DQMOffline/Muon/src/EfficiencyAnalyzer.cc

```diff
@@ -264,6 +270,9 @@ void EfficiencyAnalyzer::analyze(const edm::Event & iEvent,const edm::EventSetup
       h_allProbes_pt->Fill(muon2->pt());
       h_allProbes_eta->Fill(muon2->eta());
       h_allProbes_phi->Fill(muon2->phi());
+      h_allProbes_inner_pt->Fill(muon2->innerTrack()->innerMomentum().Rho());
+      h_allProbes_inner_eta->Fill(muon2->innerTrack()->innerPosition().Eta());
+      h_allProbes_inner_phi->Fill(muon2->innerTrack()->innerPosition().Phi());
```

Comment:
These changes have broken DQM for miniAOD jobs running separately from AOD inputs.
E.g. in  workflow 136.7611 running on 1000 events
```
   [0] Processing  Event run: 277069 lumi: 81 event: 35893405 stream: 0
   [1] Calling method for module EfficiencyAnalyzer/'LooseMuonEfficiencyAnalyzer_miniAOD'
Exception Message:
RefCore: A request to resolve a reference to a product of type 'std::vector<reco::TrackExtra>' with ProductID '2:3025'
can not be satisfied because the product cannot be found.
Probably the branch containing the product is not stored in the input file.
```

@fabozzi 
did we notice the problems already in relvals? Or are we not running these regularly?
jenkins tests will likely miss this problem because it takes 2 muons somewhat consistent with Z->mumu to make it to this problematic part of the code

Reply:
@slava77 
there workflows are only tested in IB to maintain the sequence, we do not produce them in the relval set for regular validation campaigns

Reply:
I see. So, there are no relvals which run only miniAOD with DQM included?
How are you planning to test the reminiAOD of 2016?

Reply:
Sorry for the late reply. I've been absent the whole weekend. So, if I understand correctly, these jobs are failing when asking for the innerTrack to the Muons in miniAOD jobs. I guess this is because the innerTrack is not there anymore and the reference it's not valid. I have added a "IsminiAOD" boolean in such a way that this operation is not done when we are running a miniAOD workflow. However I am unsure how to test this new code is working. Is it enough to run it over a miniAOD file? Any suggestion? Thanks a lot.

Reply:
Don't store the inner track inside pat::Muon? Just use pat::Muon::track() call instead. It will check if the track is embedded and if not, it will call innerTrack() method.

Reply:
@slava77 
> I see. So, there are no relvals which run only miniAOD with DQM included?
we do not produce relval 136.7611 (for instance) when we have a validation campaign. We could add this into the regular set (it will be needed anyway to test the fix) 
> How are you planning to test the reminiAOD of 2016?
Same as for reminiAOD of 2017: there should be a dedicated relval campaign

Reply:
> Don't we store the inner track inside pat::Muon? Just use pat::Muon::track() call instead. It will check if the track is embedded and if not, it will call innerTrack() method.

Only the track is saved, not its TrackExtra.
The TrackExtra is not available in AOD

Reply:
> I have added a "IsminiAOD" boolean in such a way that this operation is not done when we are running a miniAOD workflow. 

good 

> However I am unsure how to test this new code is working. Is it enough to run it over a miniAOD file? Any suggestion? 

In case 2017 AOD is at hand, I suggest to start from the matrix workflow 1325.51 and substitute the ttbar inputs with a Z->mumu AOD file. 
The cmsDriver command with 1K events in the test is
```cmsDriver.py step2  --conditions auto:phase1_2017_realistic -s PAT,VALIDATION:@miniAODValidation,DQM:@miniAODDQM --process PAT --era Run2_2017,run2_miniAOD_94XFall17 --eventcontent MINIAODSIM,DQM --runUnscheduled  --scenario pp --datatier MINIAODSIM,DQMIO --mc   -n 1000  --filein filelist:step1_dasquery.log --fileout file:step2.root ```

Reply:
I don't see why DQM would care about parameters at the inner point vs at point of closest approach, which is stored in Track. Also, it gets eta and phi of the inner point. Is it really what is needed? @parbol can you please clarify?

Reply:
This was an attempt to have in the DQM variables that might be sensitive to the pixels problem. Probably the same thing can be seen using the point of closest approach (what do you think Dima?), since everything we want to see is some degradation. I'm fine with both solutions: the IsminiAOD switch (already implemented), or changing to the point of closest approach.

Reply:
If it's to understand how deeply the track is going, don't you want then rho of the inner point, instead of momentum? I would expect that direction is good enough from POCA. Only location of the first hit can be interesting. Guys, what do you think?